Makefile

```diff
@@ -4,83 +4,73 @@ CC      := $(CROSS_COMPILE)gcc
 LD      := $(CROSS_COMPILE)ld
 OBJCOPY := $(CROSS_COMPILE)objcopy
 
-SDK_SRCS := \
-	SDK/Drivers/CMSIS/Device/ST/STM32L4xx/Source/Templates/gcc/startup_stm32l433xx.s \
-	SDK/Drivers/CMSIS/Device/ST/STM32L4xx/Source/Templates/system_stm32l4xx.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_adc.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_adc_ex.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_cortex.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_dma.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_gpio.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_i2c.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_i2c_ex.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_pcd.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_pcd_ex.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_pwr_ex.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_rcc.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_rcc_ex.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_tim.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_tim_ex.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_uart.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_hal_uart_ex.c \
-	SDK/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_ll_usb.c \
+TARGET ?= xildebug_big
+
+# Include target first since it chooses a platform
+include targets/$(TARGET)/build.mk
+include platforms/$(PLATFORM)/build.mk
+
+# Feature flags go here
+FEAT_POWER_PROFILER		?= 0
+FEAT_DUT_SWITCH			?= 0
+
+ifeq ($(FEAT_POWER_PROFILER),1)
+	ifeq ($(FEAT_DUT_SWITCH),0)
+	$(error FEAT_POWER_PROFILER requires FEAT_DUT_SWITCH)
+	endif
+TARGET_SRCS += \
+	app/drivers/mcp4018t.c \
+	app/power.c
+endif
+APP_CFLAGS  += -DFEAT_POWER_PROFILER=$(FEAT_POWER_PROFILER)
+
+ifeq ($(FEAT_DUT_SWITCH),1)
+TARGET_SRCS += app/drivers/max14662.c
+endif
+APP_CFLAGS += -DFEAT_DUT_SWITCH=$(FEAT_DUT_SWITCH)
+
+TARGET_SRCS += \
+	$(PLATFORM_SRCS) \
+
+TARGET_INCLUDES += \
+	targets \
+	targets/$(TARGET) \
+	platforms \
+	platforms/$(PLATFORM) \
+	$(PLATFORM_INCLUDES) \
+
+MIDDLEWARE_SRCS += \
```

Comment:
Maybe switch to an out-of-SDK freertos and just use portable's from SDK. More portable if we want to port to another chip

Reply:
Yeah the code is essentially the same

Reply:
Is that a "nah, seems pointless" or?